rust integration tests for jake to review

carols10cents opened the pull request.

shepmaster committed:
[Rust] Use generic arguments

shepmaster committed:
[Rust] No need to specify binaries in the conventional location

carols10cents committed:
[Rust] Extract conversion of EncodedData to FlightData

carols10cents committed:
[Rust] Be clearer about returned data

This function returns, potentially, a few dictionaries, and will always
return one record batch. Return these as a tuple rather than putting
them together in one vec to clarify.

carols10cents committed:
[Rust] Extract test functions to the integration testing lib to share

carols10cents committed:
[Rust] Support specifying dicts for FlightData to RecordBatch

carols10cents committed:
ARROW-8853: [Rust] [Integration Testing] Enable Flight tests

carols10cents committed:
[Rust] Remove Option that is always Some

This doesn't seem necessary

carols10cents committed:
[Rust] Skip middleware flight integration tests until tonic upgrade

Tracked in ARROW-10961. There's a bug in tonic that doesn't handle
headers and trailers correctly; it has been fixed but a new version of
tonic needs to be released and used to get the fix.

shepmaster commented on rust/integration-testing/src/bin/flight-test-integration-client.rs:
```diff
@@ -0,0 +1,439 @@
+    descriptor.cmd = b"".to_vec();
+
+    // This call is expected to fail.
+    let resp = client.get_flight_info(Request::new(descriptor.clone()));
+
+    let resp = resp.await;
```
These could probably be squished

shepmaster commented on rust/integration-testing/src/bin/flight-test-integration-client.rs:
```diff
@@ -0,0 +1,439 @@
+    Ok(())
+}
+
+fn middleware_interceptor(mut req: Request<()>) -> Result<Request<()>, Status> {
```
Was this junk leftover from my experiments, or is this needed?

carols10cents replied:
This is needed https://github.com/integer32llc/arrow/pull/4/files#diff-8c6750d1fd1e0371e4b71b357cde69fa2bd622825253108f472e21dc592793a2R79

shepmaster commented on rust/integration-testing/src/bin/flight-test-integration-server.rs:
```diff
@@ -0,0 +1,691 @@
+            .ok_or(Status::invalid_argument("Must have a descriptor"))?;
+
+        if descriptor.r#type != DescriptorType::Path as i32 || descriptor.path.is_empty()
+        {
```
There any rustfmt up in here?

carols10cents replied:
yes, rustfmt did this :P also, says the person who didn't run rustfmt on their PR :P

shepmaster commented on rust/integration-testing/src/bin/flight-test-integration-server.rs:
```diff
@@ -81,6 +83,86 @@ fn endpoint(ticket: &str, location_uri: impl Into<String>) -> FlightEndpoint {
+}
+
+// This needs to be a macro rather than `map_err(|e| tx.send(..).await).unwrap()` because
+// async closures are unstable.
```
... because `map_err` doesn't allow evaluating asynchronous code.

carols10cents replied:
obe

seddonm1 committed:
ARROW-10712: [Rust] [DataFusion] Add tests to TPC-H benchmarks

This PR adds the ability to load and parse the expected query answers included in the https://github.com/databricks/tpch-dbgen/tree/master/answers repository - which users have to clone anyway to generate the TPC-H data.

Currently DataFusion does not support Decimal types which all the numeric values in TPC-H are so there are the expected precision errors in the current results. These tests are still useful as they show some interesting results already such as non-deterministic query 5 results.

@andygrove

Closes #9015 from seddonm1/test-tpch

Authored-by: Mike Seddon <seddonm1@gmail.com>
Signed-off-by: Andy Grove <andygrove73@gmail.com>

Dandandan committed:
ARROW-11042: [Rust][DataFusion] Increase default batch size

This increases the default batch size 8x from `4096` to `32768` as it improves performance of quite some operations.
I just increased the size until performance didn't increase on my machine. Note that CSV reading also is faster on bigger batches on the bigger data sources.

This PR
```
Loading table 'part' into memory
Loaded table 'part' into memory in 125 ms
Loading table 'supplier' into memory
Loaded table 'supplier' into memory in 10 ms
Loading table 'partsupp' into memory
Loaded table 'partsupp' into memory in 381 ms
Loading table 'customer' into memory
Loaded table 'customer' into memory in 126 ms
Loading table 'orders' into memory
Loaded table 'orders' into memory in 961 ms
Loading table 'lineitem' into memory
Loaded table 'lineitem' into memory in 6382 ms
Loading table 'nation' into memory
Loaded table 'nation' into memory in 2 ms
Loading table 'region' into memory
Loaded table 'region' into memory in 2 ms
Query 12 iteration 0 took 220.2 ms
Query 12 iteration 1 took 223.2 ms
Query 12 iteration 2 took 222.4 ms
Query 12 iteration 3 took 222.2 ms
Query 12 iteration 4 took 221.8 ms
Query 12 iteration 5 took 222.0 ms
Query 12 iteration 6 took 223.1 ms
Query 12 iteration 7 took 223.7 ms
Query 12 iteration 8 took 222.5 ms
Query 12 iteration 9 took 222.9 ms
Query 12 avg time: 222.40 ms
```

Master
```
Loading table 'part' into memory
Loaded table 'part' into memory in 116 ms
Loading table 'supplier' into memory
Loaded table 'supplier' into memory in 7 ms
Loading table 'partsupp' into memory
Loaded table 'partsupp' into memory in 386 ms
Loading table 'customer' into memory
Loaded table 'customer' into memory in 115 ms
Loading table 'orders' into memory
Loaded table 'orders' into memory in 1048 ms
Loading table 'lineitem' into memory
Loaded table 'lineitem' into memory in 7673 ms
Loading table 'nation' into memory
Loaded table 'nation' into memory in 0 ms
Loading table 'region' into memory
Loaded table 'region' into memory in 0 ms
Query 12 iteration 0 took 596.1 ms
Query 12 iteration 1 took 602.0 ms
Query 12 iteration 2 took 608.1 ms
Query 12 iteration 3 took 607.9 ms
Query 12 iteration 4 took 613.5 ms
Query 12 iteration 5 took 615.3 ms
Query 12 iteration 6 took 611.6 ms
Query 12 iteration 7 took 609.8 ms
Query 12 iteration 8 took 615.7 ms
Query 12 iteration 9 took 616.9 ms
Query 12 avg time: 609.68 ms
```

Query 1 also improves a bit (but smaller improvement)

PR.
```
Query 1 iteration 0 took 653.0 ms
Query 1 iteration 1 took 653.4 ms
Query 1 iteration 2 took 652.3 ms
Query 1 iteration 3 took 658.9 ms
Query 1 iteration 4 took 655.1 ms
Query 1 iteration 5 took 662.0 ms
Query 1 iteration 6 took 659.7 ms
Query 1 iteration 7 took 662.7 ms
Query 1 iteration 8 took 669.0 ms
Query 1 iteration 9 took 665.7 ms
Query 1 avg time: 659.19 ms
```

Master:

```
Query 1 iteration 0 took 708.8 ms
Query 1 iteration 1 took 714.5 ms
Query 1 iteration 2 took 700.4 ms
Query 1 iteration 3 took 713.7 ms
Query 1 iteration 4 took 707.5 ms
Query 1 iteration 5 took 727.8 ms
Query 1 iteration 6 took 727.9 ms
Query 1 iteration 7 took 721.3 ms
Query 1 iteration 8 took 717.3 ms
Query 1 iteration 9 took 729.4 ms
Query 1 avg time: 716.85 ms
```

Closes #9021 from Dandandan/batch_size

Authored-by: Heres, Daniel <danielheres@gmail.com>
Signed-off-by: Andy Grove <andygrove73@gmail.com>

Dandandan committed:
ARROW-11046: [Rust][DataFusion] Support `count_distinct` in DataFrame API

Adds `count_distinct` function.

Closes #9028 from Dandandan/count_distinct

Authored-by: Heres, Daniel <danielheres@gmail.com>
Signed-off-by: Andy Grove <andygrove73@gmail.com>
